app/src/components/DrinkPersonalisation.js

```diff
@@ -15,6 +22,38 @@ const DrinkPersonalisation = ({ navigation, quantitySelected, setQuantitySelecte
   const [drinkPrice, setDrinkPrice] = useState(0);
   const [drinkAlcoolPercentage, setDrinkAlcoolPercentage] = useState(0);
   const [quantity, setQuantity] = useState(0);
+  const setOwnDrinksCatalog = useSetRecoilState(ownDrinksState);
+  const [doses, setDoses] = useState(0);
+  const [kCal, setkCal] = useState(0);
+  const [oldDrink, setOldDrink] = useState(null);
+  const setGlobalDrinksState = useSetRecoilState(drinksState);
+
+  const saveDrink = (name, volume, alcoolPercentage, price, icon) => {
+    setOwnDrinksCatalog((oldState) => {
+      setDoses(Math.round((alcoolPercentage * 0.8 * volume) / 10) / 10);
```

Comment:
https://react.dev/learn/queueing-a-series-of-state-updates#react-batches-state-updates